scripts/setup_sgtm_tasks_project.py

```diff
@@ -0,0 +1,228 @@
+        self.client = asana.Client.access_token(personal_access_token)
+        self.client.headers = {"Asana-Enable": "string_ids"}
+        self.workspace_id = self._get_workspace_id()
+
+    def _get_workspace_id(self) -> str:
+        """
+        Get the workspace ID that the user who created the PAT is a part of
+        """
+        pat_user = self.client.users.me()
+        if len(pat_user["workspaces"]) == 1:
+            return pat_user["workspaces"][0]["gid"]
+        else:
+            workspaces = pat_user["workspaces"]
+            workspace_list = [
+                (workspace["name"], workspace["gid"]) for workspace in workspaces
+            ]
+
+            for i, (workspace_name, workspace_id) in enumerate(workspace_list):
+                print(f"{i + 1}. {workspace_name}, {workspace_id}")
+
+            index = int(
+                input(
+                    f"Please select the workspace you want to create the SGTM project/the SGTM project exists in."
+                    f" - Enter a number between 1 and {len(workspace_list)}: "
+                )
+            )
+
+            try:
+                workspace = workspace_list[index - 1]
+                return workspace[1]
+            except IndexError as e:
+                print("Invalid selection - please try again.")
+                sys.exit(0)
+
+    def _get_user_id(self) -> str:
+        """
+        Get the user ID for the user that created the PAT
+        """
+        pat_user = self.client.users.me()
+        return pat_user["gid"]
+
+    def create_project(self, project_name: str, team_id: str) -> str:
+        """
+        Create a project with the given name, shared with the provided team in the workspace
+        """
+        create_project_data = {
+            "name": project_name,
+            "public": True,
+            "team": team_id,
+        }
+        response = self.client.projects.create_project(create_project_data)
+        return response["gid"]
+
+    def setup_custom_fields(self, project_id: str) -> None:
+        """
+        Create the custom fields that SGTM requires and add them to the given project
+        """
+        for custom_field in CUSTOM_FIELDS:
+            custom_field_data = {
+                "name": custom_field.name,
+                "enabled": True,
+                "workspace": self.workspace_id,
+                "resource_subtype": "enum",
+                "enum_options": [],
+            }
+
+            for enum_option in custom_field.enum_options:
+                custom_field_data["enum_options"].append(
+                    {
+                        "name": enum_option.name,
+                        "color": enum_option.color,
+                        "enabled": True,
+                    }
+                )
+
+            try:
+                response = self.client.custom_fields.create_custom_field(
+                    custom_field_data
+                )
+            except asana.error.PremiumOnlyError:
```

Comment:
Tested this by setting up a free user through the NUD and generating a PAT for that user